src/sinks/util/metadata.rs

```diff
@@ -1,50 +1,94 @@
 use std::num::NonZeroUsize;
 
 use vector_buffers::EventCount;
-use vector_core::{ByteSizeOf, EstimatedJsonEncodedSizeOf};
+use vector_core::{config, ByteSizeOf, EstimatedJsonEncodedSizeOf};
 
-use vector_common::{json_size::JsonSize, request_metadata::RequestMetadata};
+use vector_common::{
+    internal_event::CountByteSize,
+    json_size::JsonSize,
+    request_metadata::{GetEventCountTags, RequestCountByteSize, RequestMetadata},
+};
 
 use super::request_builder::EncodeResult;
 
 #[derive(Clone, Default)]
 pub struct RequestMetadataBuilder {
     event_count: usize,
     events_byte_size: usize,
-    events_estimated_json_encoded_byte_size: JsonSize,
+    events_estimated_json_encoded_byte_size: RequestCountByteSize,
 }
 
 impl RequestMetadataBuilder {
-    pub fn from_events<E>(events: E) -> Self
+    pub fn from_events<E>(events: &[E]) -> Self
     where
-        E: ByteSizeOf + EventCount + EstimatedJsonEncodedSizeOf,
+        E: ByteSizeOf + EventCount + GetEventCountTags + EstimatedJsonEncodedSizeOf,
     {
+        let mut size = if config::telemetry().has_tags() {
+            RequestCountByteSize::new_tagged()
+        } else {
+            RequestCountByteSize::new_untagged()
+        };
```

Comment:
Could `RequestCountByteSize` have a new method, say `new_auto` that handles this there?

Reply:
The problem is that `RequestCountByteSize` is in `vector_common` and `telemetry` is in `vector_core`.  I'll have a go at moving things around. At the very least I think I could add a method on `telemetry` to create a `RequestCountByteSize`.

Reply:
>  I'll have a go at moving things around.  

Ah, no I can't. `RequestCountByteSize` is used by `RequestMetadata` in `vector_common`. I think this was where I was at the other day when I started asking about the possibility of merging these two crates.